jdaviz/configs/default/plugins/metadata_viewer/tests/test_metadata_viewer.py

```diff
@@ -15,30 +16,32 @@ def test_view_dict(imviz_helper):
     imviz_helper.load_data(ndd_1, data_label='has_simple_meta')
     imviz_helper.load_data(ndd_2, data_label='has_nested_meta')
     imviz_helper.load_data(arr, data_label='no_meta')
-    assert mv.dc_items == ['has_simple_meta[DATA]', 'has_nested_meta[DATA]', 'no_meta']
+    assert mv.dataset.labels == ['has_simple_meta[DATA]', 'has_nested_meta[DATA]']

```

Comment:
entries without metadata are filtered out by the `DatasetSelect` component now

Reply:
Will this confuse the users when they don't see it under selection even though they loaded it?

Reply:
Also, this list does not update when metadata is added later (see snippet). I propose we don't try to be too smart here, just list everything and user sees nothing if there is no metadata, as before.

```
import numpy as np
from jdaviz import Imviz

a = np.random.random((10, 10))

imviz = Imviz()
imviz.load_data(a, data_label='test1')
imviz.app

imviz.app.data_collection[0].meta['1'] = 1  # Dropdown still shows nothing
```

Reply:
I don't think it would be confusing, but we can ask Jenn.  All of the dropdowns now filter to relevant entries, so I see this as consistent with that (it doesn't make much sense to be able to fit a model to a model, etc).  But if we want to show "no metadata for ..." and still include them in the dropdown, I can make that happen.

If updating metadata manually is something we want to support, I can try to support that... but it also was not fully supported before this PR.  Although the entry may have always shown in the dropdown, the displayed metadata would not update if you changed it manually after load, unless possibly if changing the selection.

Reply:
The problem with the notebook workflow is that we cannot predict what user wanna do with this thing. Say, they load data, and then they do some funky processing with it and decide to attach history of said processing as metadata, who are we to deny them?

Should updating metadata trigger display update? Maybe and can be future PR if requested.

Should plugin just not allow showing metadata at all if it is not there on load? I think that is a step backward.

Reply:
Do we consider accessing and modifying `app.data_collection` as user-facing?  I suspect changing items in the data collection directly will cause issues across the app as we don't listen to changes from there anywhere (aperture photometry, for example, pulls values from metadata when the dropdown changes but won't update in real-time either, etc).  I'm just worried this is a slippery slope.  But if we do want to support user changes to things within data collection, then I would prefer disabling that ability entirely until a full solution is implemented rather than relying on the user to realize the distinction between live-updates and having to toggle a selection to pull the latest values.

I think the decision of whether items without metadata should be excluded or show "no metadata" is independent though, and I don't have a terribly strong opinion except that I always personally prefer self-consistency and showing a message is actually additional logic in the plugin.